pkg/controller/direct/directbase/interfaces.go

```diff
@@ -46,5 +46,5 @@ type Adapter interface {
 
 	// Update updates an existing GCP object.
 	// This should only be called when Find has previously returned true.
-	Update(ctx context.Context) (*unstructured.Unstructured, error)
+	Update(ctx context.Context, u *unstructured.Unstructured) error
```

Comment:
just to highlight, we are expecting that implementors will use `u` to write the status to it, right?

Reply:
Right.  Specifically we write to the Status.  I think in another PR I added more docs, I'll try to track those down!